### PR TITLE
issue 924 - Fix for pattern without services

### PR DESCRIPTION
**Pull request by jiportilla**
On pattern.go

### CODE REVIEW

File: cli/exchange/pattern.go [hunk 1]
```diff
@@ -192,7 +192,11 @@ func PatternPublish(org, userPw, jsonFilePath, keyFilePath, pubKeyFilePath, patN
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, "the org specified in the input file (%s) must match the org specified on the command line (%s)", patFile.Org, org)
 	}
 	patInput := PatternInput{Label: patFile.Label, Description: patFile.Description, Public: patFile.Public, AgreementProtocols: patFile.AgreementProtocols, UserInput: patFile.UserInput}
-
+	
+	//issue 924: Patterns with no services are not allowed 
+	if patFile.Services == nil || len(patFile.Services) == 0 {
```
Comment (dabooz) [inline cli/exchange/pattern.go]: I'm not sure you are working with the latest code....I dont see msgPrinter function calls that Ling added).

Comment (jiportilla) [reply cli/exchange/pattern.go]: you are right, I rebased this branch with master, it should work now

Comment (jiportilla) [reply cli/exchange/pattern.go]: Verifying agreements: agreements still present
Sleeping for 30s giving agreements time to fail before checking again
Verifying agreements: agreements still present
Node unconfig setting is 
Unconfig loop tests are disabled.
All tests SUCCESSFUL
make: Leaving directory `/home/travis/gopath/src/github.com/open-horizon/anax/test'
The command "if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then make test -C test TEST_VARS="NOLOOP=1 TEST_PATTERNS=sall APPSTORE=1"; else make test -C test TEST_VARS="NOLOOP=1 TEST_PATTERNS=sall" && make test -C test TEST_VARS="NOLOOP=1"; fi" exited with 0.
Done. Your build exited with 0.

**Comment by jiportilla** [comment]
rebased master and re-submitted PR

File: cli/exchange/pattern.go [hunk 2]
```diff
@@ -206,7 +206,12 @@ func PatternPublish(org, userPw, jsonFilePath, keyFilePath, pubKeyFilePath, patN
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the org specified in the input file (%s) must match the org specified on the command line (%s)", patFile.Org, org))
 	}
 	patInput := PatternInput{Label: patFile.Label, Description: patFile.Description, Public: patFile.Public, AgreementProtocols: patFile.AgreementProtocols, UserInput: patFile.UserInput}
-
+	
+	//issue 924: Patterns with no services are not allowed 
+	if patFile.Services == nil || len(patFile.Services) == 0 {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("the input file (%s) must contain services, unable to proceed", patFile.Services))
```
Comment (dabooz) [inline cli/exchange/pattern.go]: please change "the input file (%s)" to "the pattern definition (%s)". Sorry I didnt notice this before.

Comment (jiportilla) [reply cli/exchange/pattern.go]: change completed --

****

**Pull request merged**